lib/improver/tests/nbhood/use_nbhood/test_ApplyNeighbourhoodProcessingWithAMask.py

```diff
@@ -0,0 +1,162 @@
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for nbhood.ApplyNeighbourhoodProcessingWithAMask."""
+
+
+import unittest
+
+import iris
+from iris.tests import IrisTest
+import numpy as np
+
+from improver.nbhood.use_nbhood import ApplyNeighbourhoodProcessingWithAMask
+from improver.tests.nbhood.nbhood.test_BaseNeighbourhoodProcessing import (
+    set_up_cube)
+
+
+def set_up_topographic_zone_cube(
+        mask_data, topographic_zone_point, topographic_zone_bounds,
+        num_time_points=1, num_grid_points=16, num_realization_points=1):
+    """Function to generate a cube with a topographic_zone coordinate. This
+    uses the existing functionality from the set_up_cube function."""
+    mask_cube = set_up_cube(
+        zero_point_indices=((0, 0, 0, 0),),
+        num_time_points=num_time_points, num_grid_points=num_grid_points,
+        num_realization_points=num_realization_points)
+    mask_cube = iris.util.squeeze(mask_cube)
+    mask_cube.data = mask_data
+    mask_cube.long_name = 'Topography mask'
+    coord_name = 'topographic_zone'
+    threshold_coord = iris.coords.AuxCoord(
+        topographic_zone_point, bounds=topographic_zone_bounds,
+        long_name=coord_name)
+    mask_cube.add_aux_coord(threshold_coord)
+    mask_cube.attributes['Topographical Type'] = "Land"
+    return mask_cube
+
+
+class Test__init__(IrisTest):
+
+    """Test the __init__ method of ApplyNeighbourhoodProcessingWithAMask."""
+
+    def test_basic(self):
+        """Test that the __init__ method returns the expected string."""
+        coord_for_masking = "topographic_zone"
+        radii = 2000
+        result = ApplyNeighbourhoodProcessingWithAMask(
+            coord_for_masking, radii)
+        msg = ("<ApplyNeighbourhoodProcessingWithAMask: coord_for_masking: "
+               "topographic_zone, neighbourhood_method: square, radii: 2000, "
+               "lead_times: None, ens_factor: 1.0, weighted_mode: True, "
+               "sum_or_fraction: fraction>")
+        self.assertEqual(str(result), msg)
+
+
+class Test__repr__(IrisTest):
+
+    """Test the __repr__ method of ApplyNeighbourhoodProcessingWithAMask."""
+
+    def test_basic(self):
+        """Test that the __repr__ method returns the expected string."""
+        coord_for_masking = "topographic_zone"
+        radii = 2000
+        result = str(ApplyNeighbourhoodProcessingWithAMask(
+            coord_for_masking, radii))
+        msg = ("<ApplyNeighbourhoodProcessingWithAMask: coord_for_masking: "
+               "topographic_zone, neighbourhood_method: square, radii: 2000, "
+               "lead_times: None, ens_factor: 1.0, weighted_mode: True, "
+               "sum_or_fraction: fraction>")
```

Comment:
Would also need updating if you add re_mask.

Reply:
Added.